cisco_firepower_connector.py

```diff
@@ -364,6 +364,13 @@ def _deploy_config(self, action_result):
             return phantom.APP_SUCCESS
 
         deployable_device_UUIDs = [device["id"] for device in self.firepower_deployable_devices]
+        filtered_devices = []
+        if len(devices) == 0:
+            filtered_devices = deployable_device_UUIDs
+        else:
+            for device in devices:
+                if device in deployable_device_UUIDs:
+                    filtered_devices.append(device)
```

Comment:
This is O(N^2) because the `in` operator against a list in python is O(N). You can get around this by using a set:

```suggestion
        deployable_device_UUIDs = {device["id"] for device in self.firepower_deployable_devices}
        if devices:
            filtered_devices = [device for device in devices if device in deployable_device_UUIDs]
        else:
            filtered_devices = list(deployable_device_UUIDs)
```